src/server/system_services/bucket_server.js

```diff
@@ -164,11 +164,11 @@ async function create_bucket(req) {
     if (req.rpc_params.namespace) {
         const read_resources = _.compact(req.rpc_params.namespace.read_resources
             .map(nsr => {
-                const res = req.system.namespace_resources_by_name[nsr.resource];
+                const res = req.system.namespace_resources_by_name && req.system.namespace_resources_by_name[nsr.resource];
                 return res && { resource: res._id, path: nsr.path };
```

Comment:
don't you need to handle it if res is undefined? 
same for other places

Reply:
not sure I follow... res? I'm checking if req.system.namespace_resources_by_name is undefined. I think you are later checking if res is defined or not in line 168.

Reply:
right, according to our discussion we are checking it later